include/wasmtime.hh

```diff
@@ -35,16 +35,73 @@
 #include <any>
 #include <initializer_list>
 #include <iostream>
+#include <limits>
 #include <memory>
 #include <optional>
-#include <span>
 #include <variant>
 #include <vector>
+#ifdef __cpp_lib_span
+#include <span>
+#endif
 
 #include "wasmtime.h"
 
 namespace wasmtime {
 
+/**
+ * \brief Span class used when c++20 is not available
+ * @tparam T Type of data
+ * @tparam Extent Static size of data refered by Span class
+ */
+template <typename T,
+          std::size_t Extent = std::numeric_limits<std::size_t>::max()>
+class Span {
+public:
+  /// \brief Type used to iterate over this span (a raw pointer)
+  using iterator = T *;
+
+  /// \brief Constructor of Span class
+  Span(T *t, std::size_t n) : ptr_{t}, size_{n} {}
+
+  /// \brief Constructor of Span class for containers
+  template <template <typename, typename> class Container>
+  Span(Container<T, std::allocator<T>> &range)
+      : ptr_{range.data()}, size_{range.size()} {}
```

Comment:
Out of curiosity, would this enable constructing a `Span` from a `std::span`?

Reply:
No, `std::span` for not have second parameter `std::allocator<T>`